facade/src/main/java/org/jboss/pnc/facade/providers/DeliverableAnalyzerReportProviderImpl.java

```diff
@@ -17,30 +17,42 @@
  */
 package org.jboss.pnc.facade.providers;
 
-import com.google.common.collect.ObjectArrays;
+import org.jboss.pnc.api.enums.LabelOperation;
+import org.jboss.pnc.dto.requests.labels.DeliverableAnalyzerReportLabelRequest;
 import org.jboss.pnc.dto.response.AnalyzedArtifact;
 import org.jboss.pnc.dto.response.Page;
 import org.jboss.pnc.facade.providers.api.DeliverableAnalyzerReportProvider;
+import org.jboss.pnc.facade.util.DeliverableAnalyzerReportLabelModifier;
+import org.jboss.pnc.facade.util.UserService;
 import org.jboss.pnc.mapper.api.ArtifactMapper;
 import org.jboss.pnc.mapper.api.DeliverableAnalyzerReportMapper;
+import org.jboss.pnc.mapper.api.UserMapper;
 import org.jboss.pnc.model.Base32LongID;
+import org.jboss.pnc.model.DeliverableAnalyzerLabelEntry;
 import org.jboss.pnc.model.DeliverableAnalyzerReport;
 import org.jboss.pnc.model.DeliverableArtifact;
 import org.jboss.pnc.spi.datastore.predicates.DeliverableArtifactPredicates;
+import org.jboss.pnc.spi.datastore.repositories.DeliverableAnalyzerLabelEntryRepository;
 import org.jboss.pnc.spi.datastore.repositories.DeliverableAnalyzerReportRepository;
 import org.jboss.pnc.spi.datastore.repositories.DeliverableArtifactRepository;
 import org.jboss.pnc.spi.datastore.repositories.api.PageInfo;
 import org.jboss.pnc.spi.datastore.repositories.api.Predicate;
 import org.jboss.pnc.spi.datastore.repositories.api.SortInfo;
 
 import javax.annotation.security.PermitAll;
-import javax.ejb.Stateless;
+import javax.annotation.security.RolesAllowed;
+import javax.enterprise.context.ApplicationScoped;
 import javax.inject.Inject;
+import java.time.Instant;
+import java.util.Date;
 import java.util.List;
 import java.util.stream.Collectors;
 
+import static org.jboss.pnc.facade.providers.api.UserRoles.SYSTEM_USER;
+import static org.jboss.pnc.spi.datastore.predicates.DeliverableAnalyzerLabelEntryPredicates.withReportId;
+
 @PermitAll
-@Stateless
+@ApplicationScoped
```

Comment:
Keep it Stateless for now, there is NCL-7590 for EJB removal. Unleas you did it to purposefully manage transactions manually.

Reply:
Done, see [37d3a94](https://github.com/project-ncl/pnc/pull/3840/commits/37d3a94f83496ac9f93d61a0ed55fb2115d85b42)